### ISSUE TITLE
Output format tree shows all unchanged resources since 0.3.8

### ISSUE THREAD

**Issue by Daniel-I-Am** [dineshba/tf-summarize#71]
This morning I updated our version of tf-summarize from 0.3.7 to 0.3.8 and the output has become very verbose in the `tree` format option.

# Example
I partially applied the following terraform code:
```terraform
resource "random_id" "a" {
  byte_length = 5
}

resource "random_id" "b" {
  byte_length = 10
}

resource "random_id" "c" {
  byte_length = 10
}
```

## Real output
```
$ terraform plan -out=tfplan
random_id.b: Refreshing state... [id=NFtC1Xo]
random_id.a: Refreshing state... [id=RJrHXAs]

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  + create
-/+ destroy and then create replacement

Terraform will perform the following actions:

  # random_id.b must be replaced
-/+ resource "random_id" "b" {
      ~ b64_std     = "NFtC1Xo=" -> (known after apply)
      ~ b64_url     = "NFtC1Xo" -> (known after apply)
      ~ byte_length = 5 -> 10 # forces replacement
      ~ dec         = "224869406074" -> (known after apply)
      ~ hex         = "345b42d57a" -> (known after apply)
      ~ id          = "NFtC1Xo" -> (known after apply)
    }

  # random_id.c will be created
  + resource "random_id" "c" {
      + b64_std     = (known after apply)
      + b64_url     = (known after apply)
      + byte_length = 10
      + dec         = (known after apply)
      + hex         = (known after apply)
      + id          = (known after apply)
    }

Plan: 2 to add, 0 to change, 1 to destroy.

───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────

Saved the plan to: tfplan

To perform exactly these actions, run the following command to apply:
    terraform apply "tfplan"
$ ~/bin/tf-summarize tfplan
+----------+-------------+
|  CHANGE  |  RESOURCE   |
+----------+-------------+
| add      | random_id.c |
+----------+-------------+
| recreate | random_id.b |
+----------+-------------+
$ ~/bin/tf-summarize -tree tfplan
|---random_id
|	|---a
|	|---b(-/+)
|	|---c(+)
```

## Expected output
The normal table view is correct, only showing changes to `random_id.b` and `random_id.c`. However, in tree view, the `random_id.a` is also (incorrectly) shown. I'd expect that output to omit `random_id.a`.

Downgrading to 0.3.7 and running the same shows the expected output:
```
$ ~/bin/tf-summarize-0.3.7 -v
Version: 0.3.7
$ ~/bin/tf-summarize-0.3.7 -tree tfplan
|---random_id
|	|---b(-/+)
|	|---c(+)
```

# Environment info
```
$ terraform version
Terraform v1.7.4
on linux_amd64
+ provider registry.terraform.io/hashicorp/random v3.6.0
$ ~/bin/tf-summarize -v
Version: 0.3.8
```

# Introduction
I ran a git bisect between 0.3.7 and 0.3.8:
```
$ cat test.sh
#!/usr/bin/env bash

set -e

go build || exit 125

(cd ../ && terraform show -json tfplan) | ./tf-summarize -tree | ( ! grep -Fq -- '---a')
$ git bisect start v0.3.8 v0.3.7
Already on 'main'
Your branch is up to date with 'origin/main'.
Bisecting: 2 revisions left to test after this (roughly 1 step)
[908b8f01702cca83cdc5e5916ab714559c86f6aa] Add nil check for IsImport (#66)
$ git bisect run ./test.sh
running './test.sh'
Bisecting: 0 revisions left to test after this (roughly 0 steps)
[b74b13da17e8c3390a56ae9e2efbcbad758b851a] [WIP] Feat: use terraform-json package (#60)
running './test.sh'
b74b13da17e8c3390a56ae9e2efbcbad758b851a is the first bad commit
```

Which points at this being introduced in #60

**Comment by dimitridittrich** [comment]
Hi Daniel.

Here at the company we started to have errors in the pipelines after this update (see below).
I resolved by returning hardcoded to version 0.3.7.


```
panic: interface conversion: interface {} is map[string]interface {}, not []uint8

goroutine *** [running]:
github.com/dineshba/tf-summarize/writer.treeValue({{0xc000***55ab0, 0x4}, 0x0, 0xc0000a70e0, {0x0, 0x0, 0x0}})
	github.com/dineshba/tf-summarize/writer/json.go:39 +0x558
github.com/dineshba/tf-summarize/writer.treeValue({{0xc0000a***0c0, 0x22}, 0x0, 0x0, {0xc000032e20, 0x***, 0x***}})
	github.com/dineshba/tf-summarize/writer/json.go:57 +0x62c
github.com/dineshba/tf-summarize/writer.treeValue({{0xc0000a***080, 0x22}, 0x0, 0x0, {0xc000032e***8, 0x***, 0x***}})
	github.com/dineshba/tf-summarize/writer/json.go:57 +0x62c
github.com/dineshba/tf-summarize/writer.treeValue({{0xc000***55a78, 0x6}, 0x0, 0x0, {0xc000***747f0, 0x2, 0x2}})
	github.com/dineshba/tf-summarize/writer/json.go:57 +0x62c
github.com/dineshba/tf-summarize/writer.JSONWriter.Write({{0xc000***747c0?, 0x2?, 0x4?}}, {0x***b9***8, 0xc000032020})
	github.com/dineshba/tf-summarize/writer/json.go:22 +0xf6
main.main()
	github.com/dineshba/tf-summarize/main.go:7*** +0x8***7
```


Just to give context, here we use it like this:
```
steps:
- task: Bash@3
  displayName: 'TF Plan Pretty Print'
  inputs:
    targetType: inline
    script: |
      find . -name plan.json > jsonplans
      COMEBACK=$(pwd)
      while read -r line; do 
          FILE=$(echo $line | sed -r 's/\.\///g' | sed -r 's/plan.json//g')
          cd $FILE
          echo "##[group] Modificações realizadas em: $FILE ========="
          docker run -v $PWD:/workspace -w /workspace ghcr.io/dineshba/tf-summarize plan.json
          docker run -v $PWD:/workspace -w /workspace ghcr.io/dineshba/tf-summarize -json plan.json > changes.json
          echo "##[endgroup]"
          cd $COMEBACK
      done < jsonplans
```

**Comment by dineshba** [comment]
Fixed with #74 and released as part of v0.3.10
